public/assets/js/filter.js

```diff
@@ -0,0 +1,21 @@
+function filter(rows, text) {
+    outter:
+        for (const row of rows) {
+            for (const column of row.children) {
+                if (column.innerText.toLowerCase().includes(text.toLowerCase())) {
+                    row.hidden = false;
+                    continue outter;
+                }
+            }
+            row.hidden = true;
+        }
+}
+
+const filterInput = document.getElementById("filter");
+const listener = function () {
+    let text = filterInput.value;
+    const rows = document.getElementsByClassName("data_row");
+    filter(rows, text);
+}
+filterInput.addEventListener("input", listener);
```

Comment:
make a not null on this:
```js
document.addEventListener('DOMContentLoaded', () => {
 // your code here
};
```
The best desicion is:
1. Create a new folder (`/modules`);
2. Make a new file (`Utility.js`);
3. There is write another function `isDOMLoaded()`.
4. Use that as a [module](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules)

Just to be in safe.

Reply:
The script tag already has the attribute defer. It will load after the HTML is parsed.